action/__tests__/checkoutRepository.test.ts

```diff
@@ -1,12 +1,27 @@
 import * as exec from '@actions/exec';
 import { context } from '@actions/github';
-import { checkoutRepository } from '../src/checkoutRepository';
+import * as checkoutRepository from '../src/checkoutRepository';
+import * as checkoutPrivateRepository from '../src/checkoutPrivateRepository';
+import * as checkoutPublicRepository from '../src/checkoutPublicRepository';
 import { describe, expect, jest, it, afterEach } from '@jest/globals';
 import { GithubEventNames } from '../src/stringEnums';
+import { ErrorStrings } from '../src/stringEnums';
+import * as githubActions from '@actions/github';
 
 describe('checkoutRepository', () => {
-  it('should checkout the pull request ref', async () => {
-    await checkoutRepository();
+  beforeEach(() => {
+    jest.restoreAllMocks();
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should checkout the pull request ref for a public repository', async () => {
+    context.eventName = GithubEventNames.PULL_REQUEST;
+
+    jest.spyOn(exec, 'exec').mockImplementation(() => Promise.resolve(0));
+
+    await checkoutRepository.checkoutRepository();
```

Comment:
nit: yeah this is kinda confusing lol

Reply:
yeah ill take note of this and whatever comes out of the bug bash ill keep this on the list